### PR TITLE
Support return types on nthArgs

### PR DESCRIPTION
**Pull request by antoniom**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. sequence
1. xor
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 24e4d75261aad79c454b11618aaabf8f95a8de79
</p>

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -1232,7 +1232,7 @@ export function nth(n: number): <T>(list: readonly T[]) => T | undefined;
 /**
  * Returns a function which returns its nth argument.
  */
-export function nthArg(n: number): (...a: readonly any[]) => any;
+export function nthArg<T>(n: number): (...a: readonly any[]) => T;
```
Comment (sheetalkamat) [inline types/ramda/index.d.ts:1235]: Dont see what the benefit of this.. 
You always have to specify the type and that can be specified as :
`R.nthArg(1)(1, 'b', false) as string` instead

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by antoniom** [comment]
Consider a function named `pipe` which accepts 2 functions as arguments (`fn1` & `fn2`): The first one accepts 2 arguments and the latter only one. 
`pipe` returns a new function that accepts 2 arguments which are applied to the `fn1` and its result is fed to `fn2` for further processing.
One could write the following typescript code to implement such a function:

```js
import * as R from "ramda";
type BiFunc<S1, S2, S3> = (arg1: S1, arg2: S2) => S3;
type Func<S1, S2> = (arg: S1) => S2;

const pipe = <S1, S2, I, O>(fn1: BiFunc<S1, S2, I>, fn2: Func<I, O>) => (
  p1: S1,
  p2: S2
) => {
  const intermediate = fn1(p1, p2);
  return fn2(intermediate);
};
```
Now calling the function as follows will lead `result` to have type assigned as `any`
```js
const result = pipe(R.nthArg(1), (x) => x + x)(1, 5); // $ExpectType any
```
Solution can be given by applying generics to `pipe` i.e.
```js
const result2 = pipe<number, number, number, number>(R.nthArg(1), (x) => x + x)(
  1,
  5
); // $ExpectType number
```

 ...or have typescript to infer the type by passing generics to `nthArgs`
```js
const result3 = pipe(R.nthArg<number>(1), (x) => x + x)(1, 5); // $ExpectType number
```

File: types/ramda/index.d.ts (hunk 1 above)
Comment (antoniom) [reply types/ramda/index.d.ts:1235]: Have a look on my comment bellow: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44768#issuecomment-632019919

**Pull request closed without merging**